source/features/restore-file.tsx

```diff
@@ -79,32 +74,47 @@ async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButt
 
 	filesRestored.add(menuItem);
 
-	menuItem.textContent = 'Restoring…';
-	event.preventDefault();
-	event.stopPropagation();
-
-	try {
-		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
+	async function restoreFile(filePath: string, progress: (message: string) => void): Promise<void> {
```

Comment:
Can you move this function to the top level? It should probably be merged with commitFileContent

Reply:
Just for clarification, making one top level function and removing both of these?

Reply:
Both what?